Swift Package Manager

Hi,

I have this issue when I try to use `GRMustache.swift` with Swift Package Manager.
[Zewo/Mustache](https://github.com/Zewo/Mustache) which is a fork of `GRMustache.swift ` seems to work with Swift Package Manager.
```
➜  MyPackage swift build
Cloning https://github.com/groue/GRMustache.swift
Using version 1.0.0 of package GRMustache.swift
Compiling Swift Module 'Mustache' (32 sources)
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Context.swift:218:18: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
    private enum Type {
                 ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Context.swift:218:18: note: backticks can escape this name if it is important to use
    private enum Type {
                 ^~~~
                 `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Context.swift:218:18: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
    private enum Type {
                 ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Context.swift:218:18: note: backticks can escape this name if it is important to use
    private enum Type {
                 ^~~~
                 `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Logger.swift:80:46: warning: '++' is deprecated: it will be removed in Swift 3
                        self.indentationLevel++
                                             ^~
                                              += 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Logger.swift:86:46: warning: '--' is deprecated: it will be removed in Swift 3
                        self.indentationLevel--
                                             ^~
                                              -= 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Logger.swift:80:46: warning: '++' is deprecated: it will be removed in Swift 3
                        self.indentationLevel++
                                             ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/Logger.swift:86:46: warning: '--' is deprecated: it will be removed in Swift 3
                        self.indentationLevel--
                                             ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateAST.swift:38:10: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
    enum Type {
         ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateAST.swift:38:10: note: backticks can escape this name if it is important to use
    enum Type {
         ^~~~
         `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateAST.swift:38:10: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
    enum Type {
         ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateAST.swift:38:10: note: backticks can escape this name if it is important to use
    enum Type {
         ^~~~
         `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateCompiler.swift:383:14: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
        enum Type {
             ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateCompiler.swift:383:14: note: backticks can escape this name if it is important to use
        enum Type {
             ^~~~
             `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateCompiler.swift:383:14: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
        enum Type {
             ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateCompiler.swift:383:14: note: backticks can escape this name if it is important to use
        enum Type {
             ^~~~
             `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateParser.swift:62:21: warning: '++' is deprecated: it will be removed in Swift 3
                    ++lineNumber
                    ^~
                                 += 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateParser.swift:77:21: warning: '++' is deprecated: it will be removed in Swift 3
                    ++lineNumber
                    ^~
                                 += 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateParser.swift:126:21: warning: '++' is deprecated: it will be removed in Swift 3
                    ++lineNumber
                    ^~
                                 += 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateParser.swift:248:21: warning: '++' is deprecated: it will be removed in Swift 3
                    ++lineNumber
                    ^~
                                 += 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateParser.swift:266:21: warning: '++' is deprecated: it will be removed in Swift 3
                    ++lineNumber
                    ^~
                                 += 1
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateToken.swift:25:10: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
    enum Type {
         ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateToken.swift:25:10: note: backticks can escape this name if it is important to use
    enum Type {
         ^~~~
         `Type`
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateToken.swift:25:10: error: type member may not be named 'Type', since it would conflict with the 'foo.Type' expression
    enum Type {
         ^
/Users/jeffreymacko/project/MyPackage/Packages/GRMustache.swift-1.0.0/Sources/TemplateToken.swift:25:10: note: backticks can escape this name if it is important to use
    enum Type {
         ^~~~
         `Type`
<unknown>:0: error: build had 1 command failures
error: exit(1): ["/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2016-02-08-a.xctoolchain/usr/bin/swift-build-tool", "-f", "/Users/jeffreymacko/project/MyPackage/.build/debug/Mustache.o/llbuild.yaml"]
```

--- a/Sources/Context.swift
+++ b/Sources/Context.swift
@@ -215,7 +215,7 @@ final public class Context {
     // =========================================================================
     // MARK: - Not public
     
-    private enum Type {
+    private enum `Type` {
         case Root
         case Box(box: MustacheBox, parent: Context)
         case PartialOverride(partialOverride: TemplateASTNode.PartialOverride, parent: Context)

--- a/Sources/TemplateAST.swift
+++ b/Sources/TemplateAST.swift
@@ -35,7 +35,7 @@ final class TemplateAST {
     // become defined.
     //
     // See TemplateRepository.templateAST(named:relativeToTemplateID:error:).
-    enum Type {
+    enum `Type` {
         case Undefined
         case Defined(nodes: [TemplateASTNode], contentType: ContentType)
     }

--- a/Sources/TemplateCompiler.swift
+++ b/Sources/TemplateCompiler.swift
@@ -380,7 +380,7 @@ final class TemplateCompiler: TemplateTokenConsumer {
             templateASTNodes.append(node)
         }
         
-        enum Type {
+        enum `Type` {
             case Root
             case Section(openingToken: TemplateToken, expression: Expression)
             case InvertedSection(openingToken: TemplateToken, expression: Expression)

--- a/Sources/TemplateToken.swift
+++ b/Sources/TemplateToken.swift
@@ -22,7 +22,7 @@
 
 
 struct TemplateToken {
-    enum Type {
+    enum `Type` {
         /// text
         case Text(text: String)
         

